### PR TITLE
[optimize](invert index) The all-hit index quickly gets the count

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

The main optimization is during the select count operation.
1. All condition fields are inverted indexes.
2. All output fields are also inverted index fields. 
3. All conditions have hit the inverted index.
In this way, when calculating the count, there is no need to read the actual data by relying on the inverted index.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: be/src/exec/olap_common.h [hunk 1]
```diff
@@ -1236,6 +1236,7 @@ Status OlapScanKeys::extend_scan_key(ColumnValueRange<primitive_type>& range,
             _end_scan_keys.emplace_back();
             _end_scan_keys.back().add_value(cast_to_string<primitive_type, CppType>(
                     range.get_range_max_value(), range.scale()));
+            *exact_value = false;
```
Comment (airborne12) [inline be/src/exec/olap_common.h:1239]: what is "exact_value" stands for?

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -155,6 +156,10 @@ class SegmentIterator::BitmapRangeIterator {
     uint32_t _buf_pos = 0;
     uint32_t _buf_size = 0;
     bool _eof = false;
+
+public:
+    uint32_t cur_ = 0;
+    roaring::Roaring bitmap_;
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:162]: we can get roaring from iterator->parent, so we do not need to store it seperately

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
ru p0

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run compile

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run compile

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run compile

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Pull request closed without merging**